### ISSUE TITLE
xc.atmos.daily_temperature_range Exception with `op` Argument

### ISSUE THREAD

**Issue by ChrisDrozdowski** [Ouranosinc/xclim#631]
* xclim version:0.22.0
* Python version: 3.8.3 64-bit
* Operating System: Win 8.1 64-bit, Win 10 64-bit

### Description
Per documentation at:
https://xclim.readthedocs.io/en/stable/indices.html#xclim.indices.daily_temperature_range

`xc.atmos.daily_temperature_range` should accept an `op` argument. However, when used:
```
da = xc.atmos.daily_temperature_range(ds.tasmax, ds.tasmin, freq='YS', op='mean')
```
There is an exception:
```
Exception has occurred: TypeError
got an unexpected keyword argument 'op'
```

**Comment by huard** [comment]
xclim has two layers: indices and indicators. indices are simple, generic functions that perform no metadata and data checks. indicators perform checks and add metadata to output values (history, etc). 

The documentation you are referring to is about indices, but the function you are using is an indicator, which wraps the indice and sets `op`: 
```python
daily_temperature_range = TasminTasmax(
    title="Mean of daily temperature range.",
    identifier="dtr",
    units="K",
    standard_name="air_temperature",
    long_name="Mean Diurnal Temperature Range",
    description="{freq} mean diurnal temperature range.",
    cell_methods="time range within days time: mean over days",
    compute=wrapped_partial(indices.daily_temperature_range, op="mean"),
)
```

The actual bug is that we are carrying over the `compute` function docstring into the indicator,  see #619.

**Comment by ChrisDrozdowski** [comment]
Thanks for addressing this so fast!

Will there be a public release soon so I can continue to develop?

**Comment by Zeitsperre** [comment]
@ChrisDrozdowski our next minor version is set for release on 2021-02-27, but we could publish a patched public release if this is addressed quickly.

The alternative is to `git clone` our master and `pip install -e .` xclim@master in your development environment to play around with it earlier. Keep an eye on notifications here to stay here in the loop!

**Comment by ChrisDrozdowski** [comment]
Well, I respect the importance of  release cycles but this issue renders your product essentially unusable for many of the indices it generates. It's not like a minor issue that can be worked around. I think an out of release cycle release would be good.

**Comment by tlogan2000** [comment]
@ChrisDrozdowski  Maybe the description provided by @huard is not sufficiently clear but I don't believe this should block you too much from using the `atmos` module going forward.  Removing the `op` parameter from the call will provide you with the mean daily temperature range via
`da = xc.atmos.daily_temperature_range(ds.tasmax, ds.tasmin, freq='YS')`

Another flavor of this same algorithm can be calculated via 
`da = xc.atmos.max_daily_temperature_range()`  in this case the `op` parameter is fixed as `'max'`

In general the `atmos` module functions as an extra layer over the `xclim.indices` algorithms providing missing data and other health checks as well as adding cf-metadata fields to the output data-array.   In this particular case we are forced to set the `op` parameter in order to match the metadata fields we wish to assign e.g, `op='mean'` with `title = "Mean of daily temperature range."` .  

I think our eventual fix will be to make sure that help info for `atmos` calls reflects the actual parameter choices available with no real change to the actual code.  FYI the number of uses of this `wrapped_partial` mechanism is relatively small I believe.   

Last, If there are flavors of `xclim.indices` not yet wrapped into the `atmos` module it is possible to customize xclim yourself (see : https://xclim.readthedocs.io/en/stable/notebooks/customize.html#Defining-new-indicators)  or more simply fall back on the use of `xclim.indices` directly but with the caveat of giving up data checks and metadata additions

**Comment by ChrisDrozdowski** [comment]
Oh goodness, I responded to the wrong Issue!!!! I am truly sorry. Multitasking never works well for me!

I meant to respond to the closed "Issue with percentile_doy #632 ". That's the one I'm pushing for a release on.

Please forgive me.
